app/models/kuroko2/job_instance.rb

```diff
@@ -43,10 +46,14 @@ def cancelable?
     tokens.first.try(:cancelable?)
   end
 
-  def cancel
+  def cancel(by:)
     self.tokens.destroy(*self.tokens)
     self.executions.destroy(*self.executions)
     self.touch(:canceled_at)
+
+    message = "This job was canceled by #{by}."
+    self.logs.warn(message)
+    Kuroko2.logger.warn(message)
   end
```

Comment:
I judged it better to keep the same as before. 
Do you have a situation of necessity?